Packages/Core/Editor/Generator/Templates/VariableTemplateFactory.cs

```diff
@@ -0,0 +1,164 @@
+using System;
+using System.Collections.ObjectModel;
+using UnityEditor;
+using UnityEngine;
+
+namespace UnityAtoms.Editor
+{
+    [InitializeOnLoad]
+    public static class VariableTemplateFactory
+    {
+        public const string atomName = "Variable";
+
+        static VariableTemplateFactory()
+        {
+            AtomGenerator.populators += AtomPopulator;
+            AtomGenerator.evaluators += AtomEvaluator;
+        }
+
+        private static void AtomPopulator(Type generatedType, System.Collections.Generic.Dictionary<string, Template> templates)
+        {
+            var atomTemplate = GetTemplate(generatedType);
+            templates.Add(atomName, atomTemplate);
+        }
+
+        private static void AtomEvaluator(Type generatedType, ReadOnlyDictionary<string, Generator.Solver> solvers)
+        {
+            var solver = solvers[atomName];
+
+            if (generatedType.IsEquatable())
+            {
+                solver.template.baseClass = typeof(EquatableAtomVariable<>).MakeGenericType(generatedType);
+
+                var body = string.Empty;
+
+                if (generatedType.IsColor())
+                {
+                    body =
+$@"
+        /// <summary>
```

Comment:
What are these methods? I don't remember them...

Reply:
They were in the original .txt templates. Under certain conditions, the generator will create methods for the generated AtomVariable as well.